kiva/fonttools/font_manager.py

```diff
@@ -670,6 +671,20 @@ def afmFontProperty(fontpath, font):
 
     return FontEntry(fontpath, name, style, variant, weight, stretch, size)
 
+def extractTTC(fpath):
+    """
+    Extract a list of ttfFontProperty objects from a ttc file
+    """
+    collection = TTCollection(str(fpath))
+
+    props = []
+    for font in collection.fonts:
+        try:
+            props.append(ttfFontProperty(fpath, font))
+        except:
```

Comment:
Bare `except`s are generally not a good idea unless you are raising immediately.  I think it is fair to say that if you can't read all the fonts from the `.ttc` file successfully it's OK to just fail for the entire file - who knows what else might fail down the line on this file.  So removing the `try` ... `except` completely here is probably the right thing.

Reply:
Yup, but then I think we need the bare `try`... `except` in `createFontList` so that the loop continues (similar to what is done in `createFontList`). At that point, trying to break this into its own function isn't seeming all that worth it. I moved this code into `createFontList` where the bare `try`...`except` may seem less bad... maybe...

Reply:
So there are two concerns here:

- first is that you should almost never to a bare `except` unless you are re-raising.  If you need to swallow exceptions you should:
   * do `except Exception` so that you _don't_ catch things like `SystemExit` and `KeyboardInterrupt`
   * log something about the error that occurred to help with debugging (why didn't my fonts load?)
- the second is that if the font file is corrupted enough that you can't read all the font, you're likely to get other issues with the fonts in the file, so I'd be tempted to just ignore everything from that file - treat the whole thing as bad.  The easiest way to do this is just to let the exception propagate and not return anything.

Reply:
I'll leave the question of further drive-by clean-up to you.  Welcome if you want to do it, but I don't want to divert effort if this PR is on a critical path for your work.

Reply:
I went ahead and replaced `verbose` with actual logging. I went ahead and did the extra clean up and replaced `verbose` with actual logging. Might as well take care of it now. 😄